service-bus.api.md

```diff
@@ -0,0 +1,256 @@
+    contentType?: string;
+    correlationId?: string;
+    label?: string;
+    messageId?: string;
+    replyTo?: string;
+    replyToSessionId?: string;
+    sessionId?: string;
+    to?: string;
+    userProperties?: any;
+}
+
+// @public
+export interface CreateBatchOptions extends OperationOptions {
+    maxSizeInBytes?: number;
+    retryOptions?: RetryOptions;
+}
+
+export { DataTransformer }
+
+// @public
+export interface DeadLetterOptions {
+    deadLetterErrorDescription: string;
+    deadLetterReason: string;
+}
+
+export { delay }
+
+export { Delivery }
+
+// @public
+export interface GetMessageIteratorOptions extends OperationOptions, WaitTimeOptions {
+}
+
+// @public
+export interface GetSessionReceiverOptions extends OperationOptions {
+    maxSessionAutoRenewLockDurationInSeconds?: number;
+    sessionId?: string;
+}
+
+// @public
+export interface MessageHandlerOptions {
+    autoComplete?: boolean;
+    maxConcurrentCalls?: number;
+    maxMessageAutoRenewLockDurationInSeconds?: number;
+}
+
+// @public
+export interface MessageHandlers<ReceivedMessageT> {
+    processError(err: Error): Promise<void>;
+    processMessage(message: ReceivedMessageT): Promise<void>;
+}
+
+export { MessagingError }
+
+// @public
+export interface ReceiveBatchOptions extends OperationOptions, WaitTimeOptions {
+}
+
+// @public
+export interface ReceivedMessage extends ServiceBusMessage {
+    readonly _amqpMessage: AmqpMessage;
+    readonly deadLetterSource?: string;
+    readonly deliveryCount?: number;
+    readonly enqueuedSequenceNumber?: number;
+    readonly enqueuedTimeUtc?: Date;
+    readonly expiresAtUtc?: Date;
+    lockedUntilUtc?: Date;
+    readonly lockToken?: string;
+    readonly sequenceNumber?: Long;
+}
+
+// @public
+export interface ReceivedMessageWithLock extends ReceivedMessage {
+    abandon(propertiesToModify?: {
+        [key: string]: any;
+    }): Promise<void>;
+    complete(): Promise<void>;
+    deadLetter(options?: DeadLetterOptions & {
+        [key: string]: any;
+    }): Promise<void>;
+    defer(propertiesToModify?: {
+        [key: string]: any;
+    }): Promise<void>;
+    renewLock(): Promise<Date>;
+}
+
+// @public
+export interface Receiver<ReceivedMessageT> {
+    close(): Promise<void>;
+    diagnostics: {
+        peek(maxMessageCount?: number): Promise<ReceivedMessage[]>;
+        peekBySequenceNumber(fromSequenceNumber: Long, maxMessageCount?: number): Promise<ReceivedMessage[]>;
+    };
+    entityPath: string;
+    getMessageIterator(options?: GetMessageIteratorOptions): AsyncIterableIterator<ReceivedMessageT>;
+    isReceivingMessages(): boolean;
+    receiveBatch(maxMessages: number, options?: ReceiveBatchOptions): Promise<ReceivedMessageT[]>;
+    receiveDeferredMessage(sequenceNumber: Long, options?: OperationOptions): Promise<ReceivedMessageT | undefined>;
+    receiveDeferredMessages(sequenceNumbers: Long[], options?: OperationOptions): Promise<ReceivedMessageT[]>;
```

Comment:
We have the single and multiple form for receiving deferred messages, but not for peek or receiveBatch. What are we using to draw the line between which features get the single & multiple variations and which get just the multiple one?